tools/apiview/parsers/CONTRIBUTING.md

```diff
@@ -0,0 +1,103 @@
+Specifically how to create or update a language parser to produce tree style tokens for APIView.
+
+## Creating Tree Style Tokens
+The main idea is to capture the hierachy of the API using a tree data structure, then maintain a flat list of tokens for each node of the tree.
+
+![APITree](APITree.svg)
+
+Each tree node has top tokens which should be used to capture the main tokens on the node, these can span multiple lines. If the language requires it use the bottom tokens to capture tokens that closes out the node, this is usually just the closing bracket and/or empty lines.
+
+- Here are the models needed
+  ```
+  object APITreeNode
+    string Name
+    string Id
+    string Kind
+    Set<string> Tags
+    Dictionary<string, string> Properties
+    List<StructuredToken> TopTokens
+    List<StructuredToken> BottomTokens
+    List<APITreeNode> Children
+
+  object StructuredToken
+    string Value
+    string Id
+    StructuredTokenKind Kind
+    Set<string> Tags
+    Dictionary<string, string> Properties 
+    Set<string> RenderClasses 
+
+  enum StructuredTokenKind
+    Content
+    LineBreak
+    NoneBreakingSpace
+    TabSpace
+    ParameterSeparator
+    Url
+  ```
+### APITreeNode
+Each module of the API (namespace, class, method) should be its own node. Members of a module (methods, in a class), (classes in a namespace) should be added as child nodes of its parent module.
+
+Sort each node at each level of the tree by your desired property, this is to ensure that difference in node order does not result in diff.
+
+Ensure each node has an Id. The combination of `Id`, `Kind` and `SubKind` should make the node unique across all nodes in the tree. This is very important. For example a class and a method can potentally have the same Id, but the kind should diffrentiate them from each other.
+
+- `Name (n)`  : The name of the tree node which will be used as label for the API Navigation. Generally use the name of the module (class, method)
+- `Id (i)` : Id of the node, which should be unique at the node level. i.e. unique among its siblings. Use whatever existing parser is assigning to DefinitionId for the main Token of the node. Each node must have an Id.
+- `Kind (k)` : What kind of node is it. Please ensure you set a Kind for each node. Using any of the following `assembly`, `class`, `delegate`, `enum`, `interface`, `method` , `namespace`, `package`, `struct`, `type` will get you the corresponding default icons for the page navigation but feel free to use something language specific.
+- `Tags (t)` : Use this for opt in or opt out boolean properties. The currently supported tags are
+  - `Deprecated` Mark a node as deprecated
+  - `Hidden`  Mark a node as Hidden
+  - `HideFromNavigation` Indicate that anode should be hidden from the page navigation.
+  - `SkipDiff` Indicate that a node should not be used in computatation of diff.
+  - `CrossLandDefId` The cross language definitionId for the node.
```

Comment:
Should live in an Enum somewhere. Also what is the actual ramification of these tags in APIView? If I mark something as deprecated, what happens in APIView? Etc. Is a `Hidden` node also hidden from navigation? I would think so. But what's the point of a hidden node? Why include a node that literally no one can ever see?

Reply:
There is a switch for toggling Hidden API display. I have converted all the currently used "magic strings" to constants